src/FrameSemantics.cc

```diff
@@ -422,6 +456,53 @@ struct ModelWrapper : public WrapperBase
       this->models.emplace_back(*_model.InterfaceModelNestedIncludeByIndex(i),
                                 *_model.InterfaceModelByIndex(i));
     }
+    for (const auto &[nestedInclude, model] : _model.MergedInterfaceModels())
+    {
+      const std::string proxyModelFrameName = computeMergedModelProxyFrameName(
+          nestedInclude->LocalModelName().value_or(model->Name()));
+
+      std::string poseRelativeTo =
+          nestedInclude->IncludePoseRelativeTo().value_or("");
+      if (poseRelativeTo.empty())
+      {
+        poseRelativeTo = "__model__";
+      }
+      this->frames.emplace_back(proxyModelFrameName,
+                                nestedInclude->IncludeRawPose().value_or(
+                                    model->ModelFramePoseInParentFrame()),
+                                poseRelativeTo, this->canonicalLinkName);
+
+      for (const auto &item : model->Links())
+      {
+        this->links.emplace_back(item.Name(), item.PoseInModelFrame(),
+                                 proxyModelFrameName);
+      }
+      for (const auto &item : model->Frames())
+      {
+        std::string attachedTo = item.AttachedTo();
+        if (item.AttachedTo() == "__model__")
+        {
+          attachedTo = proxyModelFrameName;
+        }
+        this->frames.emplace_back(item.Name(), item.PoseInAttachedToFrame(),
+                                  attachedTo, attachedTo);
+      }
+      for (const auto &item : model->Joints())
+      {
+        // TODO(azeey) In theory, the child could be "__model__" in which case,
+        // we'd have to use the proxy frame here, but not sure if "__model__" is
```

Comment:
nit "not sure" seems like it could be confirmed; worth adding a test?

(But to be fair, an unscoped `__model__` does seem like an awkward child)

Reply:
I've confirmed that `__model__` is not allowed in `//joint/child` and added a test in https://github.com/ignitionrobotics/sdformat/pull/768/commits/09b6421641b0f5051e833f83a5cf253bfbeae00a.

Reply:
Awesome!

Reply:
looking at the error messaged added to the test in https://github.com/ignitionrobotics/sdformat/commit/09b6421641b0f5051e833f83a5cf253bfbeae00a, I believe the error is raised by the [checkJointParentChildNames](https://github.com/ignitionrobotics/sdformat/blob/09b6421641b0f5051e833f83a5cf253bfbeae00a/src/parser.cc#L2555-L2564) function in `parser.cc`. To be honest, I wouldn't take this as strict proof that `__model__` is not allowed in `//joint/child`. I think our proposal is ambiguous about this, and if we modified the `checkJointParentChildNames` function to explicitly allow `__model__`, I think it would work. The same logic applies to `__model__` as a parent frame as well.

If we want to disallow `__model__` in `//joint/parent` and `//joint/child`, then let's be explicit about it

Reply:
I just tested this and found out that the logic in `checkJointParentChildNames` also prevents a scoped `__model__` from working. Since we allow `__model__` in `@relative_to` and `@attached_to`, I think it makes sense that it should be allowed in both `//joint/parent` and `//joint/child`. I think it would be best for me to create another PR making that change and update this PR afterward. Thoughts?

Reply:
https://github.com/ignitionrobotics/sdformat/pull/833

Reply:
sounds like a good plan

Reply:
Reverted https://github.com/ignitionrobotics/sdformat/commit/09b6421641b0f5051e833f83a5cf253bfbeae00a and made changes in https://github.com/ignitionrobotics/sdformat/pull/768/commits/1991fbfb73be04ced627798e8a7ed7499003003b  to account for using `__model__` in `//joint/child` in merge-included interface joints. Note, `//joint/parent` is not relevant because interface joints don't retain that information as it's not needed for building frame graphs.